### PR TITLE
XIVY-12288 display data class data as string

### PR DESCRIPTION
**Pull request by ivy-lgi**
Should we also move stuff like
- `url-helper.ts`
- `ClientContextProvider.tsx`
- `client-json-rpc.ts`
- `protocol/types.ts`
- `query-client.ts`
- etc.

to `ui-components`?

### CODE REVIEW

File: packages/dataclass-editor/src/DataClassEditor.tsx [hunk 1]
```diff
@@ -1,5 +1,46 @@
...
+  if (isPending) {
+    return (
+      <Flex alignItems='center' justifyContent='center' style={{ width: '100%', height: '100%' }}>
+        <Spinner />
+      </Flex>
+    );
+  }
+
+  if (isError) {
+    return <PanelMessage icon={IvyIcons.ErrorXMark} message={`An error has occurred: ${error.message}`} />;
```
Comment (ivy-lli) [inline packages/dataclass-editor/src/DataClassEditor.tsx:40]: maybe you need to wrap this too in a container which is 100% height? you can easily test it over the tanstack query devtools, when you trigger an error ;)

File: packages/dataclass-editor/src/DataClassEditor.tsx [hunk 2]
```diff
@@ -1,5 +1,46 @@
...
+  }
+
+  return JSON.stringify(data.data);
```
Comment (ivy-lli) [inline packages/dataclass-editor/src/DataClassEditor.tsx:43]: temp solution? 😁

File: packages/dataclass-editor/src/protocol/types.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+export type Data = { context: DataContext; data: string };
+export type DataContext = { app: string; pmv: string; file: string };
+export type EditorProps = { context: DataContext; directSave?: boolean };
+export type SaveArgs = Data & { directSave?: boolean };
+
+export interface RequestTypes {
+  data: [any, any];
+  saveData: [any, any];
```
Comment (ivy-lli) [inline packages/dataclass-editor/src/protocol/types.ts:8]: pls type this anys here

**Review by ivy-lli — APPROVED** [review]
Looks good 👍

**Review by weissreto — APPROVED** [review]

File: packages/dataclass-editor/src/DataClassEditor.tsx (hunk 1 above)
Comment (ivy-lgi) [reply packages/dataclass-editor/src/DataClassEditor.tsx:40]: Looks good. No need for it.

File: packages/dataclass-editor/src/DataClassEditor.tsx (hunk 2 above)
Comment (ivy-lgi) [reply packages/dataclass-editor/src/DataClassEditor.tsx:43]: Yes :wink:

**Pull request merged**